arduino/KnexElevator/DataAcquisitionTask.cpp

```diff
@@ -0,0 +1,117 @@
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//
+// /////////////////////////////////////////////////////////////////////////////
+
+
+// /////////////////////////////////////////////////////////////////////////////
+// Include(s)
+// /////////////////////////////////////////////////////////////////////////////
+#include <Arduino_FreeRTOS.h>
+#include <Arduino.h>
+
+#include "DataAcquisitionTask.h"
+
+
+// /////////////////////////////////////////////////////////////////////////////
+// Local Types
+// /////////////////////////////////////////////////////////////////////////////
+
+
+////////////////////////////////////////////////////////////////////////////////
+DataAcquisitionTask::DataAcquisitionTask(void)
+{
+    // Intentionally left empty
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+DataAcquisitionTask::~DataAcquisitionTask(void)
+{
+    // Intentionally left empty
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+void
+DataAcquisitionTask::Initialize(void)
+{
+    Serial.print("ENTER: ");
+    Serial.println(__PRETTY_FUNCTION__);
+
+    // Fire up our task
+    xTaskCreate(DataAcquisitionTask::RunWrapper,  // Entry Point
+                "DataAcquisitionTask",            // Task Name
+                128,                              // Stack Depth
+                this,                             // Task Parameters
+                2,                                // Priority
+                NULL);                            // Storage for task handle
+
+    Serial.print("LEAVE: ");
+    Serial.println(__PRETTY_FUNCTION__);
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+void
+DataAcquisitionTask::RunWrapper(void * pvParameters)
+{
+    // Entry point to the new task
+    DataAcquisitionTask* thisTask =
+        reinterpret_cast<DataAcquisitionTask*>(pvParameters);
+    thisTask->Run();
```

Comment:
I'm not sure that adds a great deal of value - what's the recovery plan in that case? I suppose that if we refactored this part of the thread into a base class, then we could check for null, enter an infinite loop here in the RunWrapper, blink an LED, and print the bad thread name to the serial port.  That would reduce some code in the thread that we maintain and put error handling in a decent spot (instead of spread around).

I filed a new issue to refactor this into a base class.